### PR TITLE
[Master] sync upstream

### PR DESCRIPTION
**Pull request by LukasReschke**
Another sync from upstream

### CODE REVIEW

File: apps/files_sharing/lib/API/Share20OCS.php [hunk 1]
```diff
@@ -602,13 +603,21 @@ public function updateShare($id) {
 
 			if ($newPermissions !== null &&
 				$newPermissions !== \OCP\Constants::PERMISSION_READ &&
-				$newPermissions !== (\OCP\Constants::PERMISSION_CREATE | \OCP\Constants::PERMISSION_UPDATE) &&
-				$newPermissions !== (\OCP\Constants::PERMISSION_READ | \OCP\Constants::PERMISSION_CREATE | \OCP\Constants::PERMISSION_UPDATE)) {
+				// legacy
+				$newPermissions !== (\OCP\Constants::PERMISSION_READ | \OCP\Constants::PERMISSION_CREATE | \OCP\Constants::PERMISSION_UPDATE) &&
+				// correct
+				$newPermissions !== (\OCP\Constants::PERMISSION_READ | \OCP\Constants::PERMISSION_CREATE | \OCP\Constants::PERMISSION_UPDATE | \OCP\Constants::PERMISSION_DELETE)
```
Comment (LukasReschke) [inline apps/files_sharing/lib/API/Share20OCS.php]: There was a merge conflict on L605 here.

@schiessle Please verify that I resolved it correctly. I took the ownCloud code.

**Comment by MorrisJobke** [comment]
👍

**Comment by mar1u5** [comment]
Please try to give yourself an approval :wink: @LukasReschke 
Nobody different then him is commenting to approve this PR until he has tried!!

See #238

**Comment by LukasReschke** [comment]
LGTM

**Comment by LukasReschke** [comment]
Still works 🙈

**Comment by MorrisJobke** [comment]
> Please try to give yourself an approval 😉 @LukasReschke 
> Nobody is commenting to approve this PR until he has tried!!

Maybe we need it in a PR that is based on the updated master.

**Comment by mar1u5** [comment]
Good Point. Lets try this with the next PR :unamused:

**Pull request merged**